openedx/core/djangoapps/discussions/utils.py

```diff
@@ -31,14 +32,14 @@ def get_divided_discussions(
     divided_course_wide_discussions = []
     divided_inline_discussions = []
 
-    course_wide_discussions = [topic['id'] for __, topic in course.discussion_topics.items()]
-    all_discussions = get_discussion_categories_ids(course, None, include_all=True)
-
-    for divided_discussion_id in discussion_settings.divided_discussions:
-        if divided_discussion_id in course_wide_discussions:
-            divided_course_wide_discussions.append(divided_discussion_id)
-        elif divided_discussion_id in all_discussions:
-            divided_inline_discussions.append(divided_discussion_id)
+    provider = course.discussions_settings.get('provider_type')
+    for topic_link in DiscussionTopicLink.objects.filter(
+        context_key=str(course.course_id), provider_id=provider
+    ):
+        if topic_link.enabled_in_context:
+            divided_inline_discussions.append(topic_link.external_id)
+        else:
+            divided_course_wide_discussions.append(topic_link.external_id)
```

Comment:
We need both these approaches. The first one for the legacy provider, and the second for Open edX provider. 

Also, course-wide discussions will not have a usage key, `enabled_in_context` is just for showing or hiding it in the sidebar.